streaming/src/main/scala/org/apache/spark/streaming/scheduler/JobGenerator.scala

```diff
@@ -238,13 +238,16 @@ class JobGenerator(jobScheduler: JobScheduler) extends Logging {
   /** Clear DStream metadata for the given `time`. */
   private def clearMetadata(time: Time) {
     ssc.graph.clearMetadata(time)
-    jobScheduler.receiverTracker.cleanupOldMetadata(time - graph.batchDuration)
 
     // If checkpointing is enabled, then checkpoint,
     // else mark batch to be fully processed
     if (shouldCheckpoint) {
       eventActor ! DoCheckpoint(time)
     } else {
+      // If no checkpointing to be done, delete received block information right now,
+      // else postpone until checkpoint written (better guarantees)
```

Comment:
so after talking - the part that I was confused about was how to interpret "(better guarantees)". It wasn't clear what this refers to. Maybe just remove this? I think you are just trying to say that overall, checkpointing offers better guarentees.